libc/CMakeLists.txt

```diff
@@ -706,33 +703,18 @@ if (USE_DEBUG_MALLOC)
 endif ()
 
 # This project (and its dependents) require the enclave headers and
-# library.
-target_link_libraries(oelibc PUBLIC oecore)
-
-# As a pre-build step, place the musl LIBC headers.
-add_dependencies(oelibc oelibc_includes)
+# library, and the musl headers.
+target_link_libraries(oelibc
+  PUBLIC oecore oelibc_includes
+  PRIVATE oelibasm)
```

Comment:
By using PRIVATE, would these asm objects be supplied to the linker line of the final enclave?

Reply:
This is a little tricky. Because `oelibasm` is an `OBJECT` library, the "link" step here does not do any linking, it just adds the objects of `oelibasm` to the set of objects comprising `oelibc` before the latter is linked into a static library. Once `oelibc.a` exists, `oelibasm` disappears from anything else; its objects and public/interface options are unioned with `oelibc`.

So... yes and no. Yes, because the enclave is linking against `oelibc`, but also no because the enclave never sees `oelibasm` individually.

[Here is probably a better explanation](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#object-libraries).

I don't think that `PUBLIC` and `PRIVATE` mean much when linking to an `OBJECT` library because of the way it's just unionized into the left-hand-side target.

Reply:
Auditing the exports is going to be #887.